Plugzy.API/Controllers/BaseController.cs

```diff
@@ -0,0 +1,32 @@
+using MediatR;
+
+using Microsoft.AspNetCore.Mvc;
+
+namespace Plugzy.API.Controllers;
+
+/// <Summary>
+///     Base Controller with Mediator
+/// </Summary>
+[ApiController]
+[Route("[controller]")]
+public class BaseController : ControllerBase
+{
+    /// <Summary>
+    ///     Base Controller with Mediator
+    /// </Summary>
+    protected IMediator Mediator => _mediator ??= HttpContext.RequestServices.GetService<IMediator>()!;
```

Comment:
base controller constructor ı direk DI ile alabileceğin şekilde revize edebilirsin,  her seferinde  servisleri çağırmana gerek yok.

genel olarak naming için  constructorda kullandığın ve erişilebilecek obkeler için  _mediator  gibi kullan.  yani nihai kullanılan obje adı  _xxxYyyy  gibi olsun

Reply:
Commit ile çözüldü.
https://github.com/gkhntpc/PlugzyApp/pull/14/commits/43145fe4f33ace05db8e5db63a42bc62ef88fc52